service/common/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -223,39 +212,52 @@ public static int extractHttpCodeFromCloudException(Throwable t) {
     };
   }
 
-  static int mapCloudExceptionToResponseCode(Throwable t) {
-    if (doesAnyThrowableContainAccessDeniedHint(t)) {
-      return Status.FORBIDDEN.getStatusCode();
+  /**
+   * Tries mapping a cloud exception to the HTTP code that Polaris should return
+   *
+   * @param t the throwable/exception
+   * @return the HTTP code Polaris should return, if it was possible to return a suitable mapping.
+   *     Optional.empty() otherwise.
+   */
+  static Optional<Integer> mapCloudExceptionToResponseCode(Throwable t) {
+    if (!(t instanceof S3Exception
+        || t instanceof AzureException
+        || t instanceof StorageException)) {
+      return Optional.empty();
+    }
+
+    if (containsAnyAccessDeniedHint(t.getMessage())) {
```

Comment:
I'm wondering if we shouldn't start making about of this configurable. E.g., the access denied text could be updated by config rather than hardcoding text. And the mapping of cloud vendor status codes to response codes could be useful to control by config. Doesn't have to block this PR, but it's a thought